### PR TITLE
add script to run all the storybook tests (interaction tests)

### PR DESCRIPTION
**Pull request by JulianNymark**
### Description

Useful tooling, maybe we can run this script in a github action in the future?

### try it

`yarn test:storybook`

### Change summary

- add script and related files to run storybook tests (interaction tests)

### CODE REVIEW

File: @navikt/core/react/package.json [hunk 1]
```diff
@@ -47,6 +48,7 @@
     "react-modal": "3.15.1"
   },
   "devDependencies": {
+    "@storybook/test-runner": "^0.11.0",
```
Comment (JulianNymark) [inline @navikt/core/react/package.json]: should I only add the dependency to the innermost package.json in the monorepo?

Comment (KenAJoh) [reply @navikt/core/react/package.json]: Yea, storybook-dependencies are defined @ root, so should probably be there to

**Comment by JulianNymark** [comment]
Seems like it currently visits _every_ component in a test run even when not every component has a test? (optimisation)
image

**Comment by JulianNymark** [comment]
also the test runs take a bit (took ~45 sec for me)

**Comment by KenAJoh** [comment]
https://storybook.js.org/docs/react/writing-tests/test-runner
>Storybook test runner turns all of your stories into executable tests. It is powered by [Jest](https://jestjs.io/) and [Playwright](https://playwright.dev/).

>For those [without a play function](https://storybook.js.org/docs/react/writing-stories/introduction): it verifies whether the story renders without any errors.
>For those [with a play function](https://storybook.js.org/docs/react/writing-stories/play-function): it also checks for errors in the play function and that all assertions passed.

Will run every story with and without play-tests. The 40s shouldn't be a problem if we separate this test into its own workflow. It will also need its own container since its using playwright in the background. (like our e2e workflow)
```
container:
      image: mcr.microsoft.com/playwright:v1.35.1-jammy
```

File: @navikt/core/react/.storybook/main.js
Comment (KenAJoh) [inline @navikt/core/react/.storybook/main.js:1]: Trenger vi egen config i `/react` vs den globale?

File: @navikt/core/react/package.json [hunk 2]
```diff
@@ -34,7 +34,8 @@
     "clean": "rimraf cjs esm",
     "build": "concurrently \"tsc\" \"tsc -p tsconfig.esm.json\" && yarn create-package-json-pointers-to-esm",
     "watch": "tsc --watch -p tsconfig.esm.json",
-    "test": "TZ=UTC jest"
+    "test": "TZ=UTC jest",
+    "test:storybook": "test-storybook --config-dir .storybook"
```
Comment (KenAJoh) [inline @navikt/core/react/package.json]: Denne kan kanskje kjøre på root også siden den kan teste alle storybook-stories da

File: @navikt/core/react/.storybook/main.js
Comment (JulianNymark) [reply @navikt/core/react/.storybook/main.js:1]: Hmm, jeg fikk ikke den globale til å virke 🤔

File: @navikt/core/react/package.json (hunk 2 above)
Comment (JulianNymark) [reply @navikt/core/react/package.json]:
```
➜  aksel-alt git:(main) ✗ yarn test-storybook
[test-storybook] Could not find stories in main.js in .storybook.
If you are using a mono-repository, please run the test-runner only against your sub-package, which contains a .storybook folder with "stories" defined in main.js.
You can change the config directory by using --config-dir <path-to-dir>

Error: Could not find stories in main.js in .storybook.
If you are using a mono-repository, please run the test-runner only against your sub-package, which contains a .storybook folder with "stories" defined in main.js.
You can change the config directory by using --config-dir <path-to-dir>
    at getStorybookMain (/Users/juliannymark/Repos/aksel-alt/node_modules/@storybook/test-runner/dist/util/getStorybookMain.js:50:11)
    at getStorybookMetadata (/Users/juliannymark/Repos/aksel-alt/node_modules/@storybook/test-runner/dist/util/getStorybookMetadata.js:30:61)
    at main (/Users/juliannymark/Repos/aksel-alt/node_modules/@storybook/test-runner/dist/test-storybook.js:267:98)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

er hva jeg fikk da 🤔, sliter med å finne stories, kanskje noe med stories sin matcher funksjon? (eller at det er .cjs?)

File: @navikt/core/react/.storybook/main.js
Comment (KenAJoh) [reply @navikt/core/react/.storybook/main.js:1]: Ah ser storybook anbefaler egne configs per pakke 👍

**Review by KenAJoh — DISMISSED** [review]

**Comment by KenAJoh** [comment]
Hmm kan hende det blir problematisk å bruke storybook test-runner på grunn av playwright-versjoner:
I pakken bruker de https://github.com/storybookjs/test-runner/blob/2aa338d2d14c9b78b86d511a7092071b71b14b1f/package.json#L124C1-L124C1
```json
 "playwright": "^1.14.0"
```
Problemet oppstår siden vi må definere en spesifikk versjon i container: 
```
image: mcr.microsoft.com/playwright:v1.35.1-jammy
```
Som da fører til denne feilen i e2e-workflow 
> Please install @playwright/test package before running "yarn playwright test"


Men ikke 100% her, kan hende det bare er et problem med dependency-hoisting 🤔

**Comment by KenAJoh** [comment]
@JulianNymark Hva må til for å få denne over målstreken, evt bør den lukkes?

**Comment by JulianNymark** [comment]
closing for now.

**Pull request closed without merging**